src/navsat_transform.cpp

```diff
@@ -283,15 +298,15 @@ bool NavSatTransform::datumCallback(
   transform_good_ = false;
 
   sensor_msgs::msg::NavSatFix fix;
-  fix.latitude = request.geo_pose.position.latitude;
-  fix.longitude = request.geo_pose.position.longitude;
-  fix.altitude = request.geo_pose.position.altitude;
-  fix.header.stamp = rclcpp::Time::now();
+  fix.latitude = request->geo_pose.position.latitude;
+  fix.longitude = request->geo_pose.position.longitude;
+  fix.altitude = request->geo_pose.position.altitude;
+  fix.header.stamp = rclcpp::Clock().now();
```

Comment:
use the clock in your `node_`

Reply:
Thanks, yes good catch, this could probably introduce the "different clock source error" as well.